File: rb/utility/feature_hook.rb
# Ziptastic SDK utility: feature_hook
module ZiptasticUtilities
  FeatureHook = ->(ctx, name) {
    return unless ctx.client
    features = ctx.client.features
    return unless features
    features.each do |f|
      f.send(name, ctx) if f.respond_to?(name)
    end
  }
end

File: .sdk/tm/php/utility/FeatureHook.php
<?php
declare(strict_types=1);

// Ziptastic SDK utility: feature_hook

class ZiptasticFeatureHook
{
    public static function call(ZiptasticContext $ctx, string $name): void
    {
        if (!$ctx->client) {
            return;
        }
        $features = $ctx->client->features ?? null;
        if (!$features) {
            return;
        }
        foreach ($features as $f) {
            if (method_exists($f, $name)) {
                $f->$name($ctx);
            }
        }
    }
}
